src/main/java/ch/njol/skript/expressions/ExprInventoryInfo.java

```diff
@@ -50,41 +51,42 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 	}
 
 	@Override
-	protected Object[] get(Event e) {
-		Inventory[] inventories = this.inventories.getArray(e);
-		List<Object> objects = new ArrayList<>();
+	protected Object[] get(Event event) {
+		Inventory[] inventories = this.inventories.getArray(event);
 		switch (type) {
 			case HOLDER:
+				List<InventoryHolder> holders = new ArrayList<>();
 				for (Inventory inventory : inventories) {
 					InventoryHolder holder = inventory.getHolder();
 					if (holder != null)
-						objects.add(holder);
+						holders.add(holder);
 				}
-				break;
+				return holders.toArray(new InventoryHolder[0]);
 			case ROWS:
+				List<Number> rows = new ArrayList<>();
 				for (Inventory inventory : inventories) {
 					int size = inventory.getSize();
 					if (size < 9) // Hoppers have a size of 5, we don't want to return 0
-						objects.add(1);
+						rows.add(1);
 					else
-						objects.add(size / 9);
+						rows.add(size / 9);
 				}
-				break;
+				return rows.toArray(new Number[0]);
 			case SLOTS:
+				List<Number> sizes = new ArrayList<>();
 				for (Inventory inventory : inventories) {
-					objects.add(inventory.getSize());
+					sizes.add(inventory.getSize());
 				}
-				break;
+				return sizes.toArray(new Number[0]);
 			case VIEWERS:
+				List<HumanEntity> viewers = new ArrayList<>();
 				for (Inventory inventory : inventories) {
-					objects.addAll(inventory.getViewers());
+					viewers.addAll(inventory.getViewers());
 				}
-				break;
+				return viewers.stream().filter(viewer -> viewer instanceof Player).toArray(Player[]::new);
 			default:
-				return new Object[0];
+				return (Object[]) Array.newInstance(getReturnType(), 0);
```

Comment:
this case is unexpected right? maybe we'd be better off to throw an exception for an unexpected value